### PR TITLE
Add how-to contribute

### PR DESCRIPTION
**Pull request by gtrkiller**
Add the how to contribute doc to the repository.

### CODE REVIEW

**Review by arturo-seijas — DISMISSED** [review]
LGTM

File: docs/how-to/contribute.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+```
+
+You can use the environments created by `tox` for development:
+
+```shell
+tox --notest -e unit
+source .tox/unit/bin/activate
+```
+
+### Testing
```
Comment (yanksyoon) [inline docs/how-to/contribute.md:36]: I think we could also add `src-docs` to CONTRIBUTING.md as well, [reference in Jenkins-k8s-operator](https://github.com/canonical/jenkins-k8s-operator/blob/main/CONTRIBUTING.md#generating-src-docs-for-every-commit)

**Review by yanksyoon — COMMENTED** [review]
Minor comment! Thank you! :D

File: docs/how-to/contribute.md (hunk 1 above)
Comment (gtrkiller) [reply docs/how-to/contribute.md:36]: Pollen does not have src-docs as part of its tox envs, I'll add that as well.

**Review by yanksyoon — APPROVED** [review]

File: tox.ini [hunk 2]
```diff
@@ -97,3 +97,15 @@ deps =
     -r{toxinidir}/tests/integration/requirements.txt
 commands =
     pytest -v --tb native --ignore={[vars]tst_path}unit --log-cli-level=INFO -s {posargs}
+
+[testenv:src-docs]
+allowlist_externals=sh
+setenv =
+    PYTHONPATH=src
```
Comment (cbartz) [inline tox.ini:104]:
```suggestion
    PYTHONPATH = {toxinidir}/lib:{[vars]src_path}
```
I think you need to include the libs in `PYTHONPATH`. See also [here](https://chat.canonical.com/canonical/pl/h6zp5g5sxtrttb8i99dezxzbbh).

Or don't set the PYTHONPATH and it should be passed by the `[testenv]` configuration above.

File: tox.ini [hunk 3]
```diff
@@ -97,3 +97,15 @@ deps =
...
+description = Generate documentation for src
+deps =
+    lazydocs
+    -r{toxinidir}/requirements.txt
+commands =
+    ; can't run lazydocs directly due to needing to run it on src/* which produces an invocation error in tox
+    sh generate-src-docs.sh
```
Comment (cbartz) [inline tox.ini:111]: `generate-src-docs.sh` is missing.

**Review by merkata — APPROVED** [review]
This looks great! Please add a label to the PR (like `trivial`), thanks!

**Pull request merged**